core/src/main/java/hudson/Util.java

```diff
@@ -251,19 +257,9 @@ public static void deleteFile(@Nonnull File f) throws IOException {
 
             if(!f.delete() && f.exists()) {
                 // trouble-shooting.
-                try {
-                    Class.forName("java.nio.file.Files").getMethod("delete", Class.forName("java.nio.file.Path")).invoke(null, File.class.getMethod("toPath").invoke(f));
-                } catch (InvocationTargetException x) {
-                    Throwable x2 = x.getCause();
-                    if (x2 instanceof IOException) {
-                        // may have a specific exception message
-                        throw (IOException) x2;
-                    }
-                    // else suppress
-                } catch (Throwable x) {
-                    // linkage errors, etc.; suppress
-                }
-                // see http://www.nabble.com/Sometimes-can%27t-delete-files-from-hudson.scm.SubversionSCM%24CheckOutTask.invoke%28%29-tt17333292.html
+                Files.deleteIfExists(f.toPath());
+
+                // see https://java.net/projects/hudson/lists/users/archive/2008-05/message/357
```

Comment:
this code is now in effect unreachable - if the file can not be deleted then an exception would be thrown. If the file did not exist then this loop would not be entered.If it can be deleted then this code is a no-op.

`Files.deleteIfExists(f.toPath())` can be moved to line 258 to replace `if(!f.delete() && f.exists())` and the entire `if` block.

The code works as-is so nothing major, just would seem a little strange for someone in the future.